### PR TITLE
Stage Change Exception when Stage Name includes a number

### LINKED ISSUE #9065: Stage Change Exception when Stage Name includes a number
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
An exception is thrown when trying to change the stage of contact from within the edit contact page, this issue raises when the stage name starts with a number that matches existing id of a stage

| Q   | A
| --- | ---
| Mautic version | 3.0.2
| PHP version | 7.3
| Browser | Irrelevant 

### Steps to reproduce
1. Create a stage called it `1-Open` it should have the id `1`
2. Edit on the contacts by visiting `https://site.domain/s/contacts/edit/ID`
3. Try to change the stage to the `1-Open`
4. Exception is raised
 
### Log errors
```mautic.CRITICAL: Uncaught PHP Exception Doctrine\ORM\NonUniqueResultException: "More than one result was found for query although one row or none was expected." at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php line 804 {"exception":"[object] (Doctrine\\ORM\\NonUniqueResultException(code: 0): More than one result was found for query although one row or none was expected. at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php:804)"} []```

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

### PR DESCRIPTION
**Pull request by mabumusa1**
closes https://github.com/mautic/mautic/issues/9065

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL |
| Issue(s) addressed                     | Fixes #9065 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2.

### CODE REVIEW

**Comment by npracht** [comment]
Hi @mabumusa1 why did you PR against 3.x branch ? You need to target staging :)

**Comment by mabumusa1** [comment]
Hi @npracht, 

I think I selected the wrong branch by mistake., thank you for changing to the staging branch

**Comment by mabumusa1** [comment]
@npracht can we include this in the next release, it is a minor fix

**Comment by npracht** [comment]
Yes we can but we need testers.

File: app/bundles/StageBundle/Entity/StageRepository.php [hunk 1]
```diff
@@ -214,8 +214,7 @@ public function findByIdOrName($value)
 
         if (is_numeric($value)) {
             // This is numeric value so check id and name
```
Comment (wise-gorilla) [inline app/bundles/StageBundle/Entity/StageRepository.php:216]: Edit comment accordingly

**Review by wise-gorilla — APPROVED** [review]

**Review by dennisameling — APPROVED** [review]
Makes sense from a code perspective. Might need to check @npracht's comment prior to merging this: https://github.com/mautic/mautic/issues/9065#issuecomment-674375751

**Comment by privateidaho** [comment]
Hi.  I just encountered a similar issue in 3.0.2 production.  In my implementation I group my like stages with a numerical prefix to group and sequence in the sale funnel.

In updating a contact's stage, I get this error when updating to some stages and not others. 

[2020-08-22 10:16:23] mautic.CRITICAL: Uncaught PHP Exception Doctrine\ORM\NonUniqueResultException: "More than one result was found for query although one row or none was expected." at /home/****/public_html/mautic/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php line 804 {"exception":"[object] (Doctrine\\ORM\\NonUniqueResultException(code: 0): More than one result was found for query although one row or none was expected. at /home/****/public_html/mautic/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php:804)"} [] 

SS 2020-08-22 at 10 04 02 AM

I can't update a contact record with stage '20 - FC Registered' to stage '21 - FC Attended', but can update the same record to stage '99 -%' or '25 -%'.  **I also went and changed the 'stage 21 - FC Registered' to 'FC Registered' and then tried to update the same contact and get the same error above.**

**Comment by mabumusa1** [comment]
@privateidaho the patch for this error is included in `3.1.0-rc` which is not production-ready yet, you can apply the patch or wait until the new version is released

**Comment by privateidaho** [comment]
Note that I'm still getting the same issue after updating the stage name and removing the numbers.

**Comment by mabumusa1** [comment]
May you please apply the patch and let me know if you still get the same error?

**Pull request merged**